### PR TITLE
Document state of `smt2_incremental_decision_proceduret`

### PR DESCRIPTION
**Pull request by thomasspriggs**
This PR documents the state of `smt2_incremental_decision_proceduret`.
This is useful because the state has grown large enough that saying a bit more about the purpose of each piece of state will help with maintaining it.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by TGWDB — APPROVED** [review]
Great!

File: src/solvers/smt2_incremental/smt2_incremental_decision_procedure.h [hunk 1]
```diff
@@ -70,13 +70,20 @@ class smt2_incremental_decision_proceduret final
   /// Sends the solver the definitions of the object sizes.
   void define_object_sizes();
 
+  /// Namespace for looking up the expressions which symbol_exprts relate to.
+  /// This includes the symbols defined outside of the decision procedure but
+  /// does not any SMT function identifiers the decision procedure introduces.
```
Comment (NlightNFotis) [inline src/solvers/smt2_incremental/smt2_incremental_decision_procedure.h]: Missing a verb here?

**Review by NlightNFotis — APPROVED** [review]

File: src/solvers/smt2_incremental/smt2_incremental_decision_procedure.h (hunk 1 above)
Comment (thomasspriggs) [reply src/solvers/smt2_incremental/smt2_incremental_decision_procedure.h]: I have updated this. Does it now read better?

Comment (NlightNFotis) [reply src/solvers/smt2_incremental/smt2_incremental_decision_procedure.h]: Yes :+1:

**Pull request merged**